server/src/AccessControl.hs

```diff
@@ -0,0 +1,97 @@
+{-# LANGUAGE OverloadedStrings #-}
+-- | This module defines the kinds of permissions used in icepeak and provides
+-- functions checking for sufficient permissions for certain operations.
+module AccessControl
+       ( AccessMode (..)
+       , AuthPath (..)
+       , IcepeakClaim (..)
+       , Path
+       , allowEverything
+       , accessModeToText
+       , textToAccessMode
+       , isAuthorizedByClaim
+       ) where
+
+import           Data.Aeson ((.:), (.=))
+import qualified Data.Aeson as Aeson
+import qualified Data.List  as List
+import           Data.Text  (Text)
+
+import           Store      (Path)
+
+-- * Claim datatypes
+
+-- | Defines the structure of a JWT claim for Icepeak.
+data IcepeakClaim = IcepeakClaim
+  { icepeakClaimWhitelist :: [AuthPath]
+    -- ^ The whitelist containing all authorizations.
```

Comment:
if you need this comment the name is probably not good enough. Would rename it to`icepeakClaimAcessWhitelist` make this comment die?